### PR TITLE
support toc yaml with metadata

### PR DESCRIPTION
**Pull request by hellosnow**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.TableOfContents/Utility.cs [hunk 1]
```diff
@@ -37,18 +37,36 @@ public static bool IsSupportedRelativeHref(HrefType hrefType)
                 || hrefType == HrefType.MarkdownTocFile;
         }
 
-        public static TocViewModel LoadSingleToc(string file)
+        public static TocItemViewModel LoadSingleToc(string file)
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.TableOfContents/Utility.cs]: should we return `TocViewModelWithMetadata`?

Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.TableOfContents/Utility.cs]: we will refactor in seprate review

---
In reply to: [120289039](https://github.com/dotnet/docfx/pull/1731#discussion_r120289039) [](ancestors = 120289039)

File: src/Microsoft.DocAsCode.DataContracts.Common/TocViewModelWithMetadata.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+    [Serializable]
+    public class TocViewModelWithMetadata
+    {
+        [YamlMember(Alias = "items")]
+        [JsonProperty("items")]
+        public TocViewModel Items { get; set; }
+
+        [YamlMember(Alias = "metadata")]
+        [JsonProperty("metadata")]
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.DataContracts.Common/TocViewModelWithMetadata.cs]: can this be extension?

**Comment by vicancy** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
>             // todo : metadata.

fill in metadata?

---
Refers to: src/Microsoft.DocAsCode.Build.TableOfContents/TocDocumentProcessor.cs:46 in 67d45d4. [](commit_id = 67d45d4375663226dc3c689d48ac6dcc64ff7858, deletion_comment = False)

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by hellosnow** [comment]
>             // todo : metadata.

let's fill it per requirement.

---
In reply to: [306418715](https://github.com/dotnet/docfx/pull/1731#issuecomment-306418715) [](ancestors = 306418715)

---
Refers to: src/Microsoft.DocAsCode.Build.TableOfContents/TocDocumentProcessor.cs:46 in 67d45d4. [](commit_id = 67d45d4375663226dc3c689d48ac6dcc64ff7858, deletion_comment = False)

**Pull request merged**